src/classes/CW_AudienceGroupTriggerHandlerTest.cls

```diff
@@ -0,0 +1,236 @@
+		agt.Name = 'test target';
+		agt.Is_Control_Group__c = false;	// control group
+		agt.Control_Group__c = agc.id;
+		agt.Unica_Id__c = '222333';
+		insert agt;
+		
+		insert schs;	// create schedules
+		
+		Test.stopTest();
+		
+		List<CW_Audience_Group_Schedule__c> agsC = 
+			[select id, Schedule__c, is_Control_Group__c from CW_Audience_Group_Schedule__c 
+				where Audience_Group__c = :agc.id order by Schedule__r.Start_Date__c asc];
+		List<CW_Audience_Group_Schedule__c> agsT = 
+			[select id, Schedule__c, is_Control_Group__c, Control_Group_Schedule__c from CW_Audience_Group_Schedule__c 
+				where Audience_Group__c = :agt.id order by Schedule__r.Start_Date__c asc];
+				
+		system.assertEquals(5,agsC.size());
+		system.assertEquals(5,agsT.size());
+		
+		for(Integer i=0; i<5; i++)
+		{
+			system.assert(agsC.get(i).is_Control_Group__c);
+			system.assert(!agsT.get(i).is_Control_Group__c);
+			system.assertEquals(schs.get(i).id,agsC.get(i).Schedule__c);
+			system.assertEquals(schs.get(i).id,agsT.get(i).Schedule__c);
+			system.assertEquals(agsC.get(i).id,agsT.get(i).Control_Group_Schedule__c);
+		}
+		
+	}
+	
+	static testMethod void testUpdateAGS()
+	{
+		List<CW_Schedule__c> schs = setupSchedules();	// insert later
+		
+		Test.startTest();
+		
+		// create groups with no schedules in order to avoid trigger creation
+		CW_Audience_Group__c agc = TestDataUtil.createAudienceGroup(schs.get(0).Marketing_Campaign__c);
+		agc.Name = 'test control';
+		agc.Is_Control_Group__c = true;	// control group
+		agc.Unica_Id__c = '111222';
+		insert agc;
+		
+		CW_Audience_Group__c agt = TestDataUtil.createAudienceGroup(schs.get(0).Marketing_Campaign__c);
+		agt.Name = 'test target';
+		agt.Is_Control_Group__c = false;	// control group
+		agt.Control_Group__c = null;
+		agt.Unica_Id__c = '222333';
+		insert agt;
+		
+		CW_Audience_Group__c agtNew = agt.clone(true,true);
+		agtNew.Control_Group__c = agc.id;
+		update agtNew;
+		
+		insert schs;	// create schedules
+		
+		Test.stopTest();
+		
+		List<CW_Audience_Group_Schedule__c> agsC = 
+			[select id, Schedule__c, is_Control_Group__c from CW_Audience_Group_Schedule__c 
+				where Audience_Group__c = :agc.id order by Schedule__r.Start_Date__c asc];
+		List<CW_Audience_Group_Schedule__c> agsT = 
+			[select id, Schedule__c, is_Control_Group__c, Control_Group_Schedule__c from CW_Audience_Group_Schedule__c 
+				where Audience_Group__c = :agt.id order by Schedule__r.Start_Date__c asc];
+				
+		system.assertEquals(5,agsC.size());
+		system.assertEquals(5,agsT.size());
+		
+		for(Integer i=0; i<5; i++)
+		{
+			system.assert(agsC.get(i).is_Control_Group__c);
+			system.assert(!agsT.get(i).is_Control_Group__c);
+			system.assertEquals(schs.get(i).id,agsC.get(i).Schedule__c);
+			system.assertEquals(schs.get(i).id,agsT.get(i).Schedule__c);
+			system.assertEquals(agsC.get(i).id,agsT.get(i).Control_Group_Schedule__c);
+		}
+		
+	}
+	
+	static testMethod void testDeleteAGS()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility. Style issue: methods should use the "lowerCamelCase" notation. Suggestion: rename accordingly.